### PR TITLE
bugfix/19403-tooltip-touch-follow

### LINKED ISSUE #19403: tooltip followtouchmove does not work on treemap & other similar series on mobile
Expected behaviour
tooltip.followTouchMove : true  should make the bahaviour of tooltip to follow the finger as the user swipes /drags across the plot area of heatmap/scatter/treemap/bubble series  

Actual behaviour :
Nothing happens, and the tooltip sticks to the previously selected/clicked series item .. until the user clicks another item

Live demo with steps to reproduce

try on a touch device with finger :
ive modified a simple codepen of your demo to include this setting :
https://codepen.io/anuraags/pen/GRwGjqv

Product version
Highhcharts version 11.1

### PR DESCRIPTION
**Pull request by Xillez**
Fixed #19403, tooltip didn't follow single finger touch on treemap.

---

Issue: https://github.com/highcharts/highcharts/issues/19403

During testing I found two things:
- I found that the tooltip options aren't applied for some reason, at least they're not visible in the tooltip options referenced at Pointer.ts:1347. 
- There's several demo's where touch follow isn't working and not explicitly disabled as far as I could see.

List of demos (not sure if all should have touch follow or not): 
- heatmap:
  - global-temp-heatmap
  - calendar-heatmap-month
  - highcharts/demo/heatmap
  - series-heatmap/periodic-table
  - stock/yaxis/heatmap-scrollbars
  - maps/demo/heatmap
  - maps/plotoptions/heatmap-*
- scatter:
  - 3d/scatter
  - All 3d-scatter-*
- bubble (bubble has a separate issue as far as I understood):
  - bubble-legend/color
  - packed-bubble-simple-demo
  - packed-bubble-basic

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205716154925294

### CODE REVIEW

File: ts/Series/Treemap/TreemapSeries.ts [hunk 1]
```diff
@@ -1199,6 +1201,23 @@ class TreemapSeries extends ScatterSeries {
...
+    public searchPoint(
+        e: PointerEvent,
+        compareX?: boolean
+    ): Point | undefined {
+        const series = this,
+            xAxis = series.xAxis,
+            yAxis = series.yAxis,
+            inverted = series.chart.inverted;
+        return this.searchKDTree({
+            // clientX --> plotX
```
[the remark this answers is not in the document]
Comment (KacperMadej) [reply ts/Series/Treemap/TreemapSeries.ts:1213]: Is there a nice way to make this work in TS @bre1470 ?

**Review by KacperMadej — CHANGES_REQUESTED** [review]

File: ts/Series/Treemap/TreemapSeries.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bre1470) [reply ts/Series/Treemap/TreemapSeries.ts]: The typing looks good. If you think about redundancy, we could have a simpler interface extending from `PositionObject`, but that requires refactoring.

**Comment by KacperMadej** [comment]
@Xillez 
Regarding the list of the other demos, it boils down to different series types and each one of them could be considered in a separate issue:
- `heatmap` --> it has `directTouch: true,` and the first step, at least so far, to make tooltip touch follow work is to change this to `true` and fill other missing gaps.
- `scatter3d` --> same as above. Additionally, there's a missing implementation for XYZ coordinate system in k-d-tree but it looks like something that could be added - from a brief, first look, it seems that the relevant code is extendable to accommodate the needed new feature.
- `packed-bubble-...` --> 'packedbubble` --> same as heatmap.
- bubble-legend/color --> `mapbubble` --> https://github.com/highcharts/highcharts/issues/20086 needs to be resolved first but after that there might be interference with `map` so this might not be that easy to resolve and there is no feature request pending from any user as far as I know, so maybe no need to address this at this time.

**Review by KacperMadej — APPROVED** [review]

**Review by TorsteinHonsi — COMMENTED** [review]
Thanks!

There are some issues with this approach.

* Consider a multi-level treemap like https://jsfiddle.net/pn7jwrqm/ with default options. On the desktop, the tooltip now points to items from the current level and higher levels in a mix. That's because we turned off `directTouch`. We either need to filter on the current level, or bring back `directTouch`.
* A k-d-tree search finds the closest point to the cursor. Since it only looks for one point of each rectangle in the treemap (center? corner?), we may have a situation where we hover one item and the center of another item is closer. That's why we had `directTouch` behaviour in the first place. We see this if we hover one of the elongated items in the demo.

Maybe using a k-d-tree is not the right solution in the first place to this issue. We have the touch event and a reference to the element it touches, so why look it up in a k-d-tree that is designed for finding points, not shapes?

**Comment by KacperMadej** [comment]
[The current implementation](https://github.com/highcharts/highcharts/blob/9bd53fb798301ab81c25790599256f8596a39792/ts/Series/Column/ColumnSeries.ts#L879) seems to track hover point only touchstart and proceed to [find the relevant point from the event's target](https://github.com/highcharts/highcharts/blob/9bd53fb798301ab81c25790599256f8596a39792/ts/Core/Pointer.ts#L911-L920). `touchmove` event has no updated target, so it will show the `touchstart`'s original target on `touchmove` over different elements. [Here is a potential solution](https://stackoverflow.com/a/70737325/4046096), but it might require some heavier refactoring and testing.

**Comment by KacperMadej** [comment]
If the alternative approach will work then it might be worth revisiting [this map bubble issue](https://github.com/highcharts/highcharts/issues/20086).

**Comment by TorsteinHonsi** [comment]
Setting this back to draft while we're exploring the other approach.

**Comment by Xillez** [comment]
Fixed in this PR => https://github.com/highcharts/highcharts/pull/20552, closing.

**Pull request closed without merging**